### PR TITLE
New `get-deps` command: infer PyPI dependencies from mkdocs.yml

### PR DESCRIPTION
**Pull request by oprypin**
The user story is that the following command should let you "just build" any MkDocs site:

```bash
pip install $(mkdocs get-deps) && mkdocs build
```

This cross-references 2 files:

* mkdocs.yml - `theme`, `plugins`, `markdown_extensions`
* [projects.yaml](https://github.com/mkdocs/best-of-mkdocs/blob/main/projects.yaml#L63) - a registry of all popular MkDocs-related projects and which entry points they provide - downloaded on the fly

-and prints the names of Python packages from PyPI that need to be installed to build the current MkDocs project

---

Thanks @pawamoy for inspiration and starting https://github.com/pawamoy/best-of-mkdocs

### CODE REVIEW

**Comment by pawamoy** [comment]
Thank you @oprypin, that's a great addition!

**Review by pawamoy — APPROVED** [review]

File: mkdocs/commands/get_deps.py [hunk 1]
```diff
@@ -4,31 +4,69 @@
...
+    "tag:yaml.org,2002:python/object/apply:", lambda loader, suffix, node: None
+)
+
+NotFound = ()
 
-def _extract_names(cfg, key: str) -> Iterator[str]:
-    """Get names of plugins/extensions from the config - in either a list of dicts or a dict."""
+
+def dig(cfg, keys: str):
+    """Receives a string such 'foo.bar' and returns `cfg['foo']['bar']`, or `NotFound`.
```
Comment (pawamoy) [inline mkdocs/commands/get_deps.py:43]:
```suggestion
    """Receives a string such as 'foo.bar' and returns `cfg['foo']['bar']`, or `NotFound`.
```

**Review by pawamoy — APPROVED** [review]
Missing n in dependencies (first commit)

**Comment by oprypin** [comment]
> Missing n in "dependencies" (first commit summary)

Welp, that's staying that way, that commit is very old :smiling_face_with_tear:

**Comment by pawamoy** [comment]
> [Let plugins declare their own optional dependencies](https://github.com/mkdocs/mkdocs/pull/3205/commits/7242542d1e98d87d42c509d02c1b2baeecd0a56b)
This is inferred from particular config keys
The current known plugin in need of this is mkdocstrings (handlers)

It means I can add an `extra_dependencies` key for mkdocstrings in the catalog, listing my extra/optional deps?

**Comment by oprypin** [comment]
Yes and it's already there 

https://github.com/mkdocs/catalog/blob/269c3bfb3ff9e0bb9a7b588e52e227206c914c69/projects.yaml#L258

Or did you have something else in mind?

**Comment by pawamoy** [comment]
Nice! No no, just asking.

What do the keys mean in:

```yaml
  extra_dependencies:
    plugins.mkdocstrings.handlers.crystal: mkdocstrings-crystal
    plugins.mkdocstrings.handlers.python: mkdocstrings-python
```

`plugins` to differentiate from... `themes` or `extensions` I suppose? The rest (`mkdocstrings.handlers.python`) is arbitrary?

**Comment by pawamoy** [comment]
Ah, I think I got it, it's the keys in mkdocs.yml :+1:

**Pull request merged**